smithy-codegen-core/src/main/java/software/amazon/smithy/codegen/core/directed/CodegenDirector.java

```diff
@@ -250,6 +253,22 @@ public void changeStringEnumsToEnumShapes(boolean synthesizeEnumNames) {
         });
     }
 
+    /**
+     * Sets the shapes ordering for code generation.
+     *
+     * <p>CodegenDirector order the shapes appropriately before passing them to the code generators.
+     * The default ordering is topological, and can be overridden with this method
+     *
+     * @param ordering the ordering to use for the code generation process.
+     */
+    public void setShapesGenerationOrdering(ShapesGenerationOrdering ordering) {
+        if (Objects.isNull(ordering)) {
+            this.shapesGenerationOrdering = ShapesGenerationOrdering.NONE;
```

Comment:
I'm not sure about this logic. Setting to null might imply someone wants the default behavior not necessarily NONE. There's no reason to allow setting this to null, I think. You can just not check for null here, similar to other "setter" methods and check in validateState that it is not null like the others. Also, following similar pattern to `model()`, etc. just name this method `shapesGeenerationOrdering`.